File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
       long int num =0;
        if(x<0||x>INT_MAX)
            return false;
      
        int y=x;
        while(x>0){
            num = num*10;
            if(num>INT_MAX){
                return false; }
            num = num + (x%10);
            x = x/10;
        }
        if(y==num)
        return true;
        else
            return false;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        long p=0;
        int y=x;
        if(x<0){return false;}
        while(x>0){
             p=(p*10)+(x%10);
             x=x/10;
             System.out.println(p+" "+x);
        }
        p=p+x;
        if(p==y){return true;}
        return false;
    }
}